Firmware/MVP/LTC68042.cpp

```diff
@@ -1011,10 +1045,23 @@ void LTC6804_4x20displayOFF(void)
   lcd2.noBacklight();
 }
 
-
 //---------------------------------------------------------------------------------------
 
 void LTC6804_4x20displayON(void)
 {
   lcd2.backlight();
+  lcd2.setCursor(0,0);
+  lcd2.print("LiBCM Ver. 0.0.12NM ");
+  lcd2.setCursor(0,1);
+  lcd2.print("                    ");
+  lcd2.setCursor(0,2);
+  lcd2.print("                    ");
+  lcd2.setCursor(0,3);
+  lcd2.print("                    ");
+  delay(1250);
+  lcd2.print("                    ");
+  lcd2.setCursor(0,0);
+  delay(50);
```

Comment:
This delay isn't allowed.  It will eventually cause MCM to throw a P-code.  Obviously the MVP architecture is crap, but for now I still want to add several beta features to it.  This 50 ms delay on each KEYON event is going to cause problems.

My recommendation is to only display the version string the first time LiBCM is powered on (e.g. after turning on the IMA switch).  Right now when the IMA switch is first turned on - and the USB cable plugged in (like you saw me do in Chattanooga) - the 4x20 screen just stays blank.  My proposal is that you display the LiBCM version string here instead.  Thus, you'd only be able to see the version string the first time LiBCM is powered up... but that should be good enough for debug.

Long term, our Cooperative Multitasking architecture will easily be able to handle many of the timing issues I've mentioned... but for now, these delay() functions are going to be problematic as I cram more features into the MVP super loop.

Let me know your thoughts.
If you're ok with moving the version info as proposed above, you would add this code to "LTC6804_initialize()", just after "lcd2.begin();"